From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ethan Arrowood <ethan@arrowood.dev>
Date: Mon, 7 Apr 2025 05:43:14 -0600
Subject: [PATCH 64/74] dns: restore dns query cache ttl

Fixes: https://github.com/nodejs/node/issues/57636

Co-authored-by: Robert Nagy <ronagy@icloud.com>
PR-URL: https://github.com/nodejs/node/pull/57640
Reviewed-By: Matteo Collina <matteo.collina@gmail.com>
Reviewed-By: Robert Nagy <ronagy@icloud.com>
Reviewed-By: Tim Perry <pimterry@gmail.com>
Reviewed-By: Luigi Pinca <luigipinca@gmail.com>
Reviewed-By: Ruben Bridgewater <ruben@bridgewater.de>
Reviewed-By: James M Snell <jasnell@gmail.com>
(cherry picked from commit b2b9eb36af8cc356dc96159e7565ef4d2f70777f)
---
 src/cares_wrap.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/cares_wrap.cc b/src/cares_wrap.cc
index a5be76ba8b..fa667e3af3 100644
--- a/src/cares_wrap.cc
+++ b/src/cares_wrap.cc
@@ -728,6 +728,7 @@ void ChannelWrap::Setup() {
   options.sock_state_cb_data = this;
   options.timeout = timeout_;
   options.tries = tries_;
+  options.qcache_max_ttl = 0;
 
   int r;
   if (!library_inited_) {
